spec/controllers/idv/how_to_verify_controller_spec.rb

```diff
@@ -236,17 +261,6 @@
         expect(@analytics).to have_received(:track_event).with(analytics_name, analytics_args)
       end
     end
-
-    context 'undo/back' do
-      it 'sets skip_doc_auth to nil and does not redirect' do
-        put :update, params: { undo_step: true }
```

Comment:
This is doing an HTTP PUT with `undo_step: true` as a form parameter, which does not make sense to me. There is no apparent way for the user to do this, and no code in the controller to act on this&mdash;it appears that this is equivalent to what happens when you submit a blank form, which, again, we already test.

Please let me know if I'm misinterpreting.

Reply:
I am not sure why put vs calling undo_step.call with args.  Aside from that, this test was to show that when you move backwards in the flow that some of the values on idv_session get set back to nil (the values before making any selection on this page.

Reply:
Ah-ha, this is helpful context; thanks!

I'm inclined to think that this provides equivalent coverage: https://github.com/18F/identity-idp/blob/main/spec/controllers/idv/how_to_verify_controller_spec.rb#L259-L266

I think the only way to get `undo_step` to fire are to fail the validation. Are you on board with this?

Reply:
Ummm, so I think the form submission error that you link to is checking that those items are nil because they are not set yet, not because they are set back to nil on undo. In other words, that test is viewing it as if it is someone's first time hitting this page, not that they are moving backwards. The undo step only fires on a form submission, not when folks hit the back button in the browser. So it's only called when an update action happens on a controller.

Reply:
I am not sure it captures it but think you can delete this test. I am not seeing tests for each step in the flow test undo_step but there is a spec testing undo_step so it is covered.

Reply:
The bit I linked to is testing the update action. However, I went ahead and pulled all this out into a shared context which I think will make this all easier to follow.

Just waiting on ruby 3.3.1 to build because I pulled in main. 🤦‍♂️

Reply:
That change is https://github.com/18F/identity-idp/pull/10682/commits/58fa3b873d522309e6ae33642c65d2cb203cc81f if that makes it easier to read.